starlette/datastructures.py

```diff
@@ -449,6 +451,9 @@ def _in_memory(self) -> bool:
         return not rolled_to_disk
 
     async def write(self, data: bytes) -> None:
+        if self.size is not None:
+            self.size += len(data)
```

Comment:
Hmm this is interesting. I know I just suggested making it a nullable parameter but I don't like how that creates complexity here. It makes me think that ultimately `FormParser` should not be relying on `UploadFile` to write the data; we should have a standalone `async def write_in_thread(file: IO[bytes], data: bytes) -> None`, there's just no reason to expose that in the public API and then have to deal with these sorts of weird knock-on effects.

Reply:
TBH if we make it default to 0, we should have other way to communicate to the user that size attribute is potentially not trustworthy.

Reply:
Sorry, but I don’t think I really understood what you mean

Reply:
With current implementation `size is None` means "there's no calculated information about the size of uploaded file". If we replace default `size` from `None` to `0`, then calls to `write()` will increase it, even if initial value 0 was incorrect, potentially giving people wrong (too small) impression about real size of uploaded file.

Reply:
Yeah I agree with that. So do you think what we have here is ok?

Reply:
Yes. Extra complexity is minimal, but it carries value for people dealing with file uploads.